src/commonMain/kotlin/com/charleskorn/kaml/YamlNode.kt

```diff
@@ -143,27 +143,19 @@ public data class YamlList(val items: List<YamlNode>, override val path: YamlPat
 }
 
 public data class YamlMap(val entries: Map<YamlScalar, YamlNode>, override val path: YamlPath) : YamlNode(path) {
-    init {
-        val keys = entries.keys.sortedWith { a, b ->
-            val lineComparison = a.location.line.compareTo(b.location.line)
+    private val encounteredKeys = mutableMapOf<String, YamlScalar>()
```

Comment:
Is there a reason why we wouldn't move these into the `init` block and just immediately throw the exception in there?

Reply:
If exception is thrown immediately there is there is no way to traverse the whole tree and accumulate duplicates.

However, after reading [your comment](https://github.com/charleskorn/kaml/pull/170#issuecomment-898995605) I believe it is something you'd prefer.